### PR TITLE
feat: add user details update logic

### PR DESCRIPTION
**Pull request by talshuker**

### CODE REVIEW

File: server/src/db/dal/users.dal.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    delete userDetails?.refreshTokens;
+    return userDetails;
+}
+
+export const getUserDetails = async (userId: mongoose.Types.ObjectId): Promise<Partial<IUser> | null> => {
+    const response = await UserModel.findById(userId);
+    return filteredUserDetails(response);
+}
+
+export const setUserDetails = async (userId: mongoose.Types.ObjectId, userDetails: UserDetails): Promise<Partial<IUser> | null> => {
```
Comment (MorRefael) [inline server/src/db/dal/users.dal.ts]: `userDetails: Partial<UserDetails>`

File: server/src/routes/user.routes.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import { getUserDetails, setUserDetails } from "../db/dal/users.dal";
+import { authenticate, AuthenticatedRequest } from "../auth/auth-middleware";
+import validateUpdateUserRequest from "../validation/user.validation";
+import { IUser } from "../db/schemas/user.schema";
+
+const router = Router();
+
+router.get('/', authenticate, async (req: Request, res: Response) => {
+    const { userId } = (req as AuthenticatedRequest).token;
+    let response: string | Partial<IUser> | null;
```
Comment (MorRefael) [inline server/src/routes/user.routes.ts:15]: only partial<IUser> without string

File: server/src/routes/user.routes.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    const salt = await bcrypt.genSalt(10);  
+    const userDetails = {
+        ...(email && { email }),
+        ...(username && { username }),
+        ...(gender && { gender }),
+        ...(age && { age }),
+        ...(password && { password: await bcrypt.hash(password, salt) })
+    }
+    try {
+        response = await setUserDetails(new mongoose.Types.ObjectId(userId), userDetails);
```
Comment (MorRefael) [inline server/src/routes/user.routes.ts:55]: there is no check if the user exists - new mongoose.Types.ObjectId(userId)

you can fix this by write this code.
```
const user = await getUserById(new mongoose.Types.ObjectId(userId));
if (!user) {
  res.status(HttpStatus.NOT_FOUND).json({ message: `User with id ${userId} not found` });
  return;
} 
```

File: server/src/routes/user.routes.ts (hunk 2 above)
Comment (talshuker) [reply server/src/routes/user.routes.ts:15]: response is string when error occurred

File: server/src/db/dal/users.dal.ts (hunk 1 above)
Comment (talshuker) [reply server/src/db/dal/users.dal.ts:24]: UserDetails is already defined as partial..

**Review by MorRefael — APPROVED** [review]

**Pull request merged**